src/main/java/com/epam/test_generator/controllers/CaseController.java

```diff
@@ -172,4 +166,28 @@
 
         return new ResponseEntity<>(HttpStatus.OK);
     }
+
+    @ApiOperation(value = "Perform status changing event on given case", nickname = "performEvent")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "OK", response = Status.class),
+        @ApiResponse(code = 400, message = "Can't perform given event on current case")
+    })
+    @ApiImplicitParams({
+        @ApiImplicitParam(name = "suitId", value = "ID of suit which contains the case",
+            required = true, dataType = "long", paramType = "path"),
+        @ApiImplicitParam(name = "caseId", value = "IDs of cases to perform event on",
+            required = true, dataType = "long", paramType = "path"),
+        @ApiImplicitParam(name = "event", value = "Name of event to be performed",
+            required = true, dataType = "String", paramType = "path")
+    })
+    @RequestMapping(value = "suits/{suitId}/cases/{caseId}/events/{event}", method = RequestMethod.PUT,
+        consumes = "application/json")
+    public ResponseEntity<Status> performEvent(@PathVariable("suitId") long suitId,
+                                               @PathVariable("caseId") long caseId,
+                                               @PathVariable("event") Event event)
+        throws Exception {
```

Comment:
This exception is never thrown.

Reply:
method from state machine interfaces requiers throwing of this exception